### ISSUE TITLE
Integration der von der Wetterstation gesendeten Daten der lokalen Wetterstation über einen in der eigenen Anwendung zur Verfügung gestellten Endpunkt in der  API

### ISSUE THREAD

**Issue by saschadoemer** [vitrum-connect/5gla-api#223]
**Beschreibung**

Als Entwickler möchte ich die von der Wetterstation gesendeten Daten der lokalen Wetterstation über einen in der eigenen Anwendung zur Verfügung gestellten Endpunkt in die API integrieren, um die aktuellen Wetterinformationen direkt in unserem System zu nutzen.

**Akzeptanzkriterien**

Damit die User Story erfolgreich umgesetzt ist, wird erwartet ...

- ... dass die gesendeten Daten der Wetterstation über den spezifischen Endpunkt empfangen werden.
- ... dass die empfangenen Daten korrekt in der Anwendung verarbeitet und gespeichert werden.
- ... dass die Wetterdaten im definierten Format für die Anwendung bereitgestellt werden.
- ... dass Fehler und Ausfälle beim Empfang der Daten entsprechend behandelt werden.
- 

**Funktionale Anforderungen**

- Implementierung eines Endpunkts zur Entgegennahme der Daten von der Wetterstation.
- Verarbeitung und Speicherung der empfangenen Wetterdaten in der Datenbank.
- Bereitstellung der Wetterdaten im definierten Format für die Anwendung.
- Implementierung von Fehlerbehandlung und Logmanagement für den Datenempfang.
- Durchführung von Tests zur Sicherstellung der korrekten Datenverarbeitung und -speicherung.

**Nicht-funktionale Anforderungen**

- Sicherstellung der hohen Verfügbarkeit und Zuverlässigkeit des Endpunkts.
- Implementierung von Sicherheitsmaßnahmen zur Authentifizierung und Autorisierung.
- Minimierung der Latenzzeiten beim Empfang und der Verarbeitung der Daten.
- Dokumentation der Endpunktfunktionalität und der Datenverarbeitungsprozesse.
- Gewährleistung der Skalierbarkeit bei einer steigenden Anzahl an Datenübertragungen.

**Comment by saschadoemer** [comment]
@Chromeggs ja, via spezieller API können die Daten für eine konkrete (eigene) Wetterstation abgefragt werden. Findet sich hier in der [Dokumentation](https://openweathermap.org/stations). Wollen wir dann taktisch auf die Integration via OpenWeather umstellen? D.h. das Registrieren einer Station mit der entsprechenden ID über einen Endpunkt und Anlage der Station via API. Dann müssen wir "nur" noch auf der Wetterstation den API-Key von OpenWeather hinterlegen und von dort aus die OpenWeather-Plattform bespielen. Von der Komplexität ändert sich das Verhalten kaum, nur der Kanal für die Daten ändert sich.

**Comment by Chromeggs** [comment]
@saschadoemer 
Äh ja, äh nein, äh vielleicht 
Ich glaube ich verstehe den Vorschlag nicht...
🤪
Was ich gut fände: unsere Plattform kann 
a) sowohl (auf Tenant-Basis) auf die allgemeinen Wetterdaten von OW (auf Basis deren Modell) für eine bestimmte Region zugreifen, 
b) als auch spezifische Daten aus einer odere mehrerer privater Wetterestationen, die vorher bei OW eingebunden wurden, für einen Tenant wieder ausgeben.  
Macht das Sinn?
Ist es das, was du meinst?

**Comment by saschadoemer** [comment]
Der erste von dir angesprochene Teil ist in der User Story #222 abgebildet, d.h. das wird hier nicht betrachtet. Hier ist nur der Teil der lokalen Wetterstation relevant. Und da gilt - kann angebunden werden und kann auch direkt über deren API abgefragt werden.

**Comment by saschadoemer** [comment]
@DomenikINBW lt. @FelixSchmidt23 gibt es ein Repo, welches die Anbindung der lokalen Wetterstation bereits in Arbeit hat. Kannst du das hier verlinken?

**Comment by saschadoemer** [comment]
@DomenikINBW kannst du mir die Excel-Datei für den manuellen Import anhängen.

**Comment by DomenikINBW** [comment]
[Messdaten-8-August-2024.xlsx](https://github.com/user-attachments/files/17160210/Messdaten-8-August-2024.xlsx)

Einmal die Daten von August als Beispiel.

**Comment by saschadoemer** [comment]
@DomenikINBW das ist auch das Format, wie es später direkt eingespielt werden soll?

**Comment by DomenikINBW** [comment]
@saschadoemer genau so kommt es aus der Wetterstation. 
Bei den älteren Daten von vor ein paar Jahren sieht die Tabelle noch etwas anders aus. Da würde ich es aber vorschlagen das vor dem einspielen auf das aktuelle Format zu parsen.

**Comment by saschadoemer** [comment]
Integration gestartet, erste Implementierung für das Parsen der Excel fertiggestellt.

**Comment by saschadoemer** [comment]
Bis Ende der Woche verschoben, wg. der Unterstützung für die Visualisierung.
